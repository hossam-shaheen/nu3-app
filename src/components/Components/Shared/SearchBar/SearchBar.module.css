.search-bar {
  position: relative;
  width: 100%;
}
