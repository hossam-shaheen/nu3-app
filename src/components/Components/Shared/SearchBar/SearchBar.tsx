import { FunctionComponent, useState, ChangeEvent } from 'react';
import { ProductsSearchProps } from 'src/interfaces/interfaces';
import classes from './SearchBar.module.css';

const SearchBar: FunctionComponent<{ onSearch: ProductsSearchProps["onSearch"] }> = ({ onSearch }): JSX.Element => {
    const [searchResult, setSearchResult] = useState('');

    const onSearchChange = (e: ChangeEvent<HTMLInputElement>): void => {
        const searchKeyWord = e.target.value;
        setSearchResult(searchKeyWord);
        onSearch(searchKeyWord);
    }

    return (<div className={classes["search-bar"]}>
        <input
            type="text"
            placeholder="Search for a product"
            value={searchResult}
            onChange={onSearchChange}
        />


    </div>);

}
export default SearchBar;
