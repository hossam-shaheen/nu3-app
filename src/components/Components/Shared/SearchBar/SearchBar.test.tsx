import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SearchBar from './SearchBar';

test('SearchBar displayed in the document : Renders', () => {
    render(<BrowserRouter>
        <SearchBar onSearch={jest.fn()} />
    </BrowserRouter>);
    const searchInput = screen.getByPlaceholderText('Search for a product');;
    expect(searchInput).toBeInTheDocument();
});

test(`SearchBar didn't Type product name : Renders`, () => {
    render(<BrowserRouter>
        <SearchBar onSearch={jest.fn()} />
    </BrowserRouter>);
    const searchInput = screen.getByPlaceholderText('Search for a product');;
    fireEvent.change(searchInput, { target: { value: '' } })
    const searchInputValue = searchInput.getAttribute('value');
    expect(searchInputValue).toBe('');
});


test('SearchBar Type product name : Renders', () => {
    render(<BrowserRouter>
        <SearchBar onSearch={jest.fn()} />
    </BrowserRouter>);
    const searchInput = screen.getByPlaceholderText('Search for a product');;
    fireEvent.change(searchInput, { target: { value: 'nu3' } })
    const searchInputValue = searchInput.getAttribute('value');
    expect(searchInputValue).toBe('nu3');
});

test('SearchBar displayed : Snapshot', () => {
    const component = renderer.create(<BrowserRouter>
        <SearchBar onSearch={jest.fn()} />
    </BrowserRouter>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});