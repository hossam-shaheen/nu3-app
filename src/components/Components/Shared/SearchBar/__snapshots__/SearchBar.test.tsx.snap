// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar displayed : Snapshot 1`] = `
<div
  className="search-bar"
>
  <input
    onChange={[Function]}
    placeholder="Search for a product"
    type="text"
    value=""
  />
</div>
`;
